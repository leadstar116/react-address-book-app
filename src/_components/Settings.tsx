import React from 'react'
import { Row, Col } from 'react-bootstrap'
import { useDispatch, useSelector } from 'react-redux'
import { initialState } from '../_constants/state.interface'
import { updateNationality } from '../_helpers/settings.thunk'

const Settings = () => {
    const dispatch = useDispatch()
    const settings = useSelector((state: initialState) => state.settingsReducer)

    return (
        <div className="p-2">
            <h2 className="text-center">Settings</h2>
            <div className="settings-container">
                <Row className="text-center">
                    <Col xs="12" sm="3">Nationality: </Col>
                    <Col xs="12" sm="9">
                        <select
                            name="nationality"
                            id="nationality"
                            className="nationality-select"
                            value={settings.location.nationality}
                            onChange={(e) => dispatch(updateNationality(e.target.value))}
                        >
                            <option value="Any">Any</option>
                            <option value="CH">CH</option>
                            <option value="ES">ES</option>
                            <option value="FR">FR</option>
                            <option value="GB">GB</option>
                        </select>
                    </Col>
                </Row>
            </div>
        </div>
    )
}

export default Settings