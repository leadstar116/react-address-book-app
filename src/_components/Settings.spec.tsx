import configureStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { LocationEnum, LocationSettings } from '../_constants/settings.interface';
import Settings from './Settings'

const mockStore = configureStore();

const initialState = {
    location: {
        nationality: LocationEnum.Any
    } as LocationSettings
}

describe('Settings Component', () => {
    let store
    let component: ReactTestRenderer

    store = mockStore({
        settingsReducer: initialState
    })

    store.dispatch = jest.fn();

    component = renderer.create(
        <Provider store={store}>
            <Settings />
        </Provider>
    );

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})