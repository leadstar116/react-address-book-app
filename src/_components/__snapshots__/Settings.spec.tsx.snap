// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Component should render with given state from Redux store 1`] = `
<div
  className="p-2"
>
  <h2
    className="text-center"
  >
    Settings
  </h2>
  <div
    className="settings-container"
  >
    <div
      className="text-center row"
    >
      <div
        className="col-sm-3 col-12"
      >
        Nationality: 
      </div>
      <div
        className="col-sm-9 col-12"
      >
        <select
          className="nationality-select"
          id="nationality"
          name="nationality"
          onChange={[Function]}
          value=""
        >
          <option
            value="Any"
          >
            Any
          </option>
          <option
            value="CH"
          >
            CH
          </option>
          <option
            value="ES"
          >
            ES
          </option>
          <option
            value="FR"
          >
            FR
          </option>
          <option
            value="GB"
          >
            GB
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
